contracts/lupi.sol

```diff
@@ -15,17 +20,35 @@ contract lupi is owned {
     State state = State.Betting;
 
     Ticket[] tickets;
-    uint revealedCount;
+    uint revealedCount; // TODO: shall we replace this with a getter returning revealedBets.length?
 
     // bet => tickets
     mapping(uint => uint[]) revealedTickets;
     uint[] revealedBets;
 
     uint winningTicket;
 
-    function lupi() {
+    // TODO: function getBets(address _player) constant returns bets[]
+
+    function lupi(uint _requiredBetAmount, uint _ticketCountLimit, uint _feePt ) {
+        requiredBetAmount = _requiredBetAmount;
+        ticketCountLimit = _ticketCountLimit;
+        feePt = _feePt;
         // ticket zero is reserved
-        tickets.push(Ticket(0, 0, 0, 0));
+        tickets.push(Ticket(0, 0, 0, 0)); // TODO: shall we replace this for a more readable logic?
```

Comment:
Not a big fan either. The space saving used to make more sense when I had a different data structure. I'll fix this by introducing two distinct states instead of `Closed`, eg. `Won` and `Tied`